src/main/kotlin/org/rust/cargo/project/settings/ui/RustProjectSettingsPanel.kt

```diff
@@ -146,9 +160,35 @@ class RustProjectSettingsPanel(
     private val RsToolchainBase.rustup: Rustup? get() = rustup(cargoProjectDir)
 }
 
-private fun String.blankToNull(): String? = if (isBlank()) null else this
+private fun String.blankToNull(): String? = ifBlank { null }
 
 private fun wrapComponent(component: JComponent): JComponent =
     JPanel(BorderLayout()).apply {
         add(component, BorderLayout.NORTH)
     }
+
+/**
+ * Obtains a list of toolchains on a pool using [toolchainObtainer], then fills [toolchainComboBox] on the EDT.
+ */
+@Suppress("UnstableApiUsage")
+private fun addToolchainsAsync(
+    toolchainComboBox: RsToolchainPathChoosingComboBox,
+    toolchainObtainer: () -> List<Path>
+) {
+    toolchainComboBox.setBusy(true)
+    ApplicationManager.getApplication().executeOnPooledThread {
+        var toolchains = emptyList<Path>()
+        try {
+            toolchains = toolchainObtainer()
+        } finally {
+            val executor = AppUIExecutor.onUiThread(ModalityState.any()).expireWith(toolchainComboBox)
```

Comment:
It seems it doesn't work because you never dispose `toolchainComboBox`. So I see two possible variants:
- dispose `RsToolchainPathChoosingComboBox` object in `org.rust.cargo.project.settings.ui.RustProjectSettingsPanel#dispose`
- or pass another disposable